### PR TITLE
Fix: The enableFeedback property of InkWell cannot be set to a nullab…

### PR DESCRIPTION
**Pull request by StanleyCocos**
*I found that enableFeedback can be set to null, but in actual use, setting it to null has no effect and is overwritten by true. I think this is an issue.*

```dart
class InkWell extends InkResponse {
  const InkWell({
    super.key,
    super.child,
    super.onTap,
    super.onDoubleTap,
    super.onLongPress,
    super.onTapDown,
    super.onTapUp,
    super.onTapCancel,
    super.onSecondaryTap,
    super.onSecondaryTapUp,
    super.onSecondaryTapDown,
    super.onSecondaryTapCancel,
    super.onHighlightChanged,
    super.onHover,
    super.mouseCursor,
    super.focusColor,
    super.hoverColor,
    super.highlightColor,
    super.overlayColor,
    super.splashColor,
    super.splashFactory,
    super.radius,
    super.borderRadius,
    super.customBorder,
    bool? enableFeedback = true,
    super.excludeFromSemantics,
    super.focusNode,
    super.canRequestFocus,
    super.onFocusChange,
    super.autofocus,
    super.statesController,
    super.hoverDuration,
  }) : super(
    containedInkWell: true,
    highlightShape: BoxShape.rectangle,
    enableFeedback: enableFeedback ?? true,
  );
}
```

Call Error
```dart
InkWell(
    enableFeedback: null,
)
```

### CODE REVIEW

**Comment by TahaTesser** [comment]
@StanleyCocos 
Thanks for the contribution! We will need tests in order to land this.

File: packages/flutter/lib/src/material/ink_well.dart
Comment (nate-thegrate) [inline packages/flutter/lib/src/material/ink_well.dart:1461]: Technically this could be a breaking change for someone, if they've explicitly called `InkWell(enableFeedback: null)`. I do think this is a nice readability improvement though.

File: packages/flutter/lib/src/material/button_style_button.dart
Comment (nate-thegrate) [inline packages/flutter/lib/src/material/button_style_button.dart:505]: nit: I think it'd be better to add `?? true` to the declaration site for `resolvedEnableFeedback`.

**Review by nate-thegrate — COMMENTED** [review]
Looks great overall, but @TahaTesser is right—this PR will need either a test or a test exemption. (One could argue that there are virtually no semantic changes here, but in my experience, requesting a test exemption for a borderline case hasn't succeeded.)

One approach could be to keep the `bool? enableFeedback = true` for now and ~~add an assert; then test whether the assert fires~~ (perhaps this isn't the best idea, since we'd have to ignore [tighten_type_of_initializing_formals](https://dart.dev/lints/tighten_type_of_initializing_formals)), and then a while after it lands in stable we could get back in here and finish the job.

File: packages/flutter/lib/src/material/ink_well.dart
Comment (StanleyCocos) [reply packages/flutter/lib/src/material/ink_well.dart:1465]: I’m not entirely sure how this should be adjusted. Should I mark this PR as a breaking change? It doesn't seem like it's significant enough for that, but I'm unsure. Could you provide some guidance? Thank you!

**Comment by StanleyCocos** [comment]
@nate-thegrate If you could help me resolve the robot comments above, it would be greatly appreciated. Thank you!

**Comment by nate-thegrate** [comment]
Sounds good! I should have an update for you within the next few days :)

**Comment by StanleyCocos** [comment]
I'm very happy to see this outcome. Thank you

**Comment by stuartmorgan** [comment]
test-exempt: code refactor with no semantic change

**Review by nate-thegrate — APPROVED** [review]

**Comment by stuartmorgan** [comment]
> I believe that if someone explicitly calls InkWell(enableFeedback: null), it is an error.

Interestingly, at the time this file was migrated to null-safety and the `bool?` was introduced, the constructor had the `?? true` handling of `null` already, but also had a comment saying:

```dart
  /// The [mouseCursor], [enableFeedback], and [excludeFromSemantics] arguments
  /// must not be null.
```

**Comment by nate-thegrate** [comment]
Interesting, I guess this must have slipped under the radar!

@StanleyCocos we'll be able to merge this pull request once it has 2 approvals.

**Review by bleroux — APPROVED** [review]
LGTM! Thanks for the cleanup. 🙏

**Pull request merged**
